File: tests/fortran/test_hello.f90
! test_hello.f90 - Tests for the Fortran hello_mod module
program test_hello
    use hello_mod, only: greet_str
    implicit none

    character(len=64) :: result
    integer :: failures

    failures = 0

    ! Test basic greeting
    result = greet_str("World")
    if (trim(result) /= "Hello, World!") then
        write(*, '("FAIL: greet_str(World) expected ''Hello, World!'' got ''", A, "''")') trim(result)
        failures = failures + 1
    end if

    ! Test with a different name
    result = greet_str("CMake")
    if (trim(result) /= "Hello, CMake!") then
        write(*, '("FAIL: greet_str(CMake) expected ''Hello, CMake!'' got ''", A, "''")') trim(result)
        failures = failures + 1
    end if

    if (failures == 0) then
        write(*, '("All Fortran hello tests passed.")')
    end if

    if (failures > 0) stop 1

end program test_hello

File: tests/cpp/test_hello.cpp
// test_hello.cpp - Unit tests for the C++ hello library
#include <cassert>
#include <iostream>
#include "hello/hello.hpp"

int main()
{
    // Test basic greeting
    std::string result = hello::greet("World");
    assert(result == "Hello, World!" && "greet(\"World\") should return \"Hello, World!\"");

    // Test with a different name
    std::string result2 = hello::greet("CMake");
    assert(result2 == "Hello, CMake!" && "greet(\"CMake\") should return \"Hello, CMake!\"");

    // Test with an empty string
    std::string result3 = hello::greet("");
    assert(result3 == "Hello, !" && "greet(\"\") should return \"Hello, !\"");

    std::cout << "All C++ hello tests passed.\n";
    return 0;
}
